File: juegos1000tres-backend/src/main/java/com/juegos1000tres/juegos1000tres_backend/comunicacion/Conexion.java
package com.juegos1000tres.juegos1000tres_backend.comunicacion;

public interface Conexion<PAYLOAD> {

    String TIPO_COMPROBAR_HOMOLOGO = "ComprobarHomologo";

    void conectar();

    void desconectar();

    void enviar(PAYLOAD payload);

    PAYLOAD recibir();

    Class<PAYLOAD> getClasePayload();

    String getTipoComunicacion();

    default String getTipoVerificacion() {
        return TIPO_COMPROBAR_HOMOLOGO;
    }

    default boolean comprobarHomologo(String tipoComunicacionFront) {
        if (tipoComunicacionFront == null || tipoComunicacionFront.isBlank()) {
            return false;
        }

        return getTipoComunicacion().equalsIgnoreCase(tipoComunicacionFront.trim());
    }
}

File: juegos1000tres-frontend-miguel/src/comunicacion/Conexion.ts
export abstract class Conexion<PAYLOAD> {
  static readonly TIPO_COMPROBAR_HOMOLOGO = "ComprobarHomologo";

  abstract conectar(): Promise<void>;

  abstract desconectar(): Promise<void>;

  abstract enviar(payload: PAYLOAD): Promise<void>;

  abstract recibir(): Promise<PAYLOAD>;

  abstract getTipoPayload(): string;

  abstract getTipoComunicacion(): string;

  getTipoVerificacion(): string {
    return Conexion.TIPO_COMPROBAR_HOMOLOGO;
  }

  comprobarHomologo(tipoComunicacionFront: string): boolean {
    if (!tipoComunicacionFront || !tipoComunicacionFront.trim()) {
      return false;
    }

    return this.getTipoComunicacion().toLowerCase() === tipoComunicacionFront.trim().toLowerCase();
  }
}
